### DISCUSSION TITLE
Is it possible to render figures from R code chunks using the LaTeX figure (\begin{figure}) environment directly?

### DISCUSSION THREAD

**Discussion by Niklewa** [quarto-dev/quarto-cli#6736] [Q&A] [answered]
### Description

I have a project written in an Rmd file where the chunks that display visualizations in the document look like this. Unfortunately, this does not work when used in Quarto. I am familiar with the '#|' way of specifying figure characteristics. The best solution for me would not require modifying all the chunks. Is it possible to work with the pure LaTeX figure environment?

\begin{figure}

```{r}

grid.arrange(FairPlot,UniformPlot, TwoPlot, TwoUnbalancedPlot)
```

\label{fig-theFigure}
\caption{caption of the figure}
\end{figure}

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by Niklewa** [reply]
Thanks for the answer. Here an example with improper render: 


````qmd
---
title: "A document"
format: pdf
---

```{r, echo= FALSE}

thePlot = plot(cars)
```

\begin{figure}

```{r , echo= FALSE, eval=TRUE, fig.align = "center", out.width = "80%", message = FALSE, warning = FALSE, results = FALSE}

thePlot
```

\label{fig:cars}
\caption{A caption}
\end{figure}
````


The image will not be properly assigned to the figure environment, the troubling result is visible in the latex render (includegraphics should be inside of the environment):


\includegraphics{REp_files/figure-latex/unnamed-chunk-1-1.pdf}

\begin{figure}


::: {.cell layout-align="center"}

:::


\label{fig:cars}
\caption{A caption}
\end{figure}

**Reply by cderv** [reply, marked as answer]
You can't do that with Quarto. It may have worked with R Markdown and knitr but probaby as a side effect because in some cases knitr will generate some LaTeX code directly (which will be mixed with yours). 

here Quarto will always include image using Markdown syntax `![caption](path)` So your are mixing markdown with LaTeX. 

Quarto has all the tool you need to produce a figure environment with caption and cross reference from a chunk. You just need to use the right syntax. 

````markdown
```{r}
#| label: fig-cars
#| echo: false
#| eval: true
#| fig-align: center
#| out-width: 80%
#| message: false
#| warning: false
#| results: false
plot(cars)
```
````
will produce 

````latex
\begin{figure}

{\centering \includegraphics[width=0.8\textwidth,height=\textheight]{index_files/figure-pdf/fig-cars-1.pdf}

}

\caption{\label{fig-cars}\textbf{?(caption)}}

\end{figure}
````

I realize this is some more typing for you, but this is the Quarto syntax for figure. Otherwise, you need to customize your tooling (knitr hooks, and Lua filter), but I would not advice that for this usage, and it would probably be longer that converting your doc (manually or creating a converter)

**Reply by Niklewa** [reply]
Thank you for help!
